tremor-script/src/interpreter/imut_expr.rs

```diff
@@ -181,21 +208,13 @@ where
 
         if let Some(target_map) = target_value.as_object() {
             // Record comprehension case
+
             value_vec.reserve(target_map.len());
-            // NOTE: Since we we are going to create new data from this
-            // object we are cloning it.
-            // This is also required since we might mutate. If we restruct
-            // mutation in the future we could get rid of this.
-
-            'comprehension_outer: for (k, v) in target_map.clone() {
-                stry!(set_local_shadow(
-                    self,
-                    local,
-                    &env.meta,
-                    expr.key_id,
-                    Value::String(k)
-                ));
-                stry!(set_local_shadow(self, local, &env.meta, expr.val_id, v));
+
+            'comprehension_outer: for (k, v) in target_map {
+                let k = Value::String(k.clone());
```

Comment:
using `Value::from()` is the more idiomatic approach here as it avoids the internal representation
```suggestion
                let k = Value::form(k.clone());
```